libs/designer-ui/src/lib/templates/templateparameters/templateparametersField.tsx

```diff
@@ -0,0 +1,146 @@
+
+  const intl = useIntl();
+
+  const parameterDetails: TemplateParameterFieldDetails = {
+    name: `${definition.name}-${NAME_KEY}`,
+    value: `${definition.name}-${VALUE_KEY}`,
+    description: `${definition.name}-${DESCRIPTION_KEY}`,
+    type: `${definition.name}-${TYPE_KEY}`,
+  };
+
+  const nameTitle = intl.formatMessage({
+    defaultMessage: 'Name',
+    id: 'm8Q61y',
+    description: 'Parameter Field Name Title',
+  });
+  const typeTitle = intl.formatMessage({
+    defaultMessage: 'Type',
+    id: 'tNoZx2',
+    description: 'Parameter Field Type Title',
+  });
+  const descriptionTitle = intl.formatMessage({
+    defaultMessage: 'Description',
+    id: 'UXDOiw',
+    description: 'Parameter Field Description Title',
+  });
+  const valueTitle = intl.formatMessage({
+    defaultMessage: 'Value',
+    id: 'ClZW2r',
+    description: 'Parameter Field Value Title',
+  });
+  const valueDescription = intl.formatMessage({
+    defaultMessage: 'Enter value for parameter.',
+    id: 'rSIBjh',
+    description: 'Parameter Field Value Placeholder Text',
+  });
+
+  const onValueChange = (_event: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>, newValue?: string): void => {
+    handleValueChange(newValue);
+  };
+
+  const handleValueChange = (value?: string) => {
+    setValue(value);
+
+    onChange?.({
+      name: definition.name,
+      newDefinition: { ...definition, value },
+    });
+  };
+
+  return (
+    <>
+      <div className="msla-workflow-parameter-field">
+        <Label styles={labelStyles} required={false} htmlFor={parameterDetails.name}>
+          {nameTitle}
+        </Label>
+        <Text className="msla-workflow-parameter-read-only">{definition.name}</Text>
+      </div>
+      <div className="msla-workflow-parameter-field">
+        <Label styles={labelStyles} required={false} htmlFor={parameterDetails.type}>
+          {typeTitle}
+        </Label>
+        <Text className="msla-workflow-parameter-read-only">{definition.type}</Text>
+      </div>
+      <div className="msla-workflow-parameter-field">
+        <Label styles={labelStyles} required={false} htmlFor={parameterDetails.description}>
+          {descriptionTitle}
+        </Label>
+        <Text className="msla-workflow-parameter-read-only">{definition?.description}</Text>
+      </div>
+      <div className="msla-workflow-parameter-field">
+        <Label styles={labelStyles} required={definition?.required} htmlFor={parameterDetails.value}>
+          {valueTitle}
+        </Label>
+        <TextField
+          data-testid={parameterDetails.value}
+          id={parameterDetails.value}
+          ariaLabel={valueTitle}
+          placeholder={valueDescription}
+          value={value}
+          errorMessage={validationError}
+          styles={textFieldStyles}
+          onChange={onValueChange}
+          disabled={false}
+        />
+      </div>
+    </>
+  );
+};
+
+function stringifyValue(value: any): string {
```

Comment:
Maybe move this function to the utils or any library so we can reuse it. As I can see there the same in workflowparametersField.tsx, and maybe add a couple of unit tests to it